app/uk/gov/hmrc/cataloguefrontend/service/DefaultBranchesService.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cataloguefrontend.service
+
+import uk.gov.hmrc.cataloguefrontend.connector.RepositoryDisplayDetails
+import uk.gov.hmrc.http.HeaderCarrier
+
+import javax.inject._
+import scala.concurrent.ExecutionContext
+import scala.language.postfixOps
+
+@Singleton
+class DefaultBranchesService @Inject()()(implicit val ec: ExecutionContext){
+
+  def allTeams(repos: Seq[RepositoryDisplayDetails]): Seq[String] = {
+    val repositories = repos.map(t => t.teamNames)
+    repositories.flatten.distinct.sorted
+  }
+
+  def updatedDefaultBranchCount(
+      repos:            Seq[RepositoryDisplayDetails],
+      name:             Option[String],
+      defaultBranch:    Option[String],
+      teamNames:        Option[String],
+      singleOwnership:  Option[Boolean],
+      archived:         Option[Boolean]): Int = {
+
+    var results: Seq[RepositoryDisplayDetails] = repos
+
+    if(name.isDefined)              { results = results.filter(_.name contains name.get) }
+    if(defaultBranch.isDefined)     { results = results.filter(_.defaultBranch contains defaultBranch.get) }
+    if(teamNames.isDefined)         { results = results.filter(_.teamNames contains teamNames.get) }
+    if(singleOwnership.isDefined)   { results = results.filter(_.teamNames.length == 1) }
+    if(archived.isEmpty)            { results = results.filter(_.isArchived == false) }
```

Comment:
are these filters exclusive? if so then `else if` might be better than multiple ifs. 
If they're not exclusive is `results` just being overwritten each time?

Reply:
They're exclusive - This is because each search field on the page triggers a separate if statement. `results` is being overwritten each time still.

Reply:
what if `archived = Some(true)` - shouldn't it be `filter(_.isArchived == archived.get)`?
By creating a pattern, this can be simplified too.